packages/dds/tree/src/codec/discriminatedUnions.ts

```diff
@@ -80,7 +80,7 @@ export const unionOptions: ObjectOptions = {
  * ```
  * where only a single property of `EncodedOperation` is populated for a given encoded value.
  */
-export class DiscriminatedUnionDispatcher<TUnion extends object, TArgs extends any[], TResult> {
+export class DiscriminatedUnionDispatcher<TUnion extends object, TArgs extends unknown[], TResult> {
```

Comment:
Function argument arrays is one of the only cases I know of where `any` is actually more correct than `unknown`. Any chance there's a sub-rule exception for this?

See the first topic here:
https://www.totaltypescript.com/any-considered-harmful

There's a bunch of similar changes at the bottom of this PR as well.

Reply:
Not an exact match, but overlapping: https://typescript-eslint.io/rules/no-explicit-any/#ignorerestargs

I'll give it a shot and see if it makes things better or worse than rule disables.

Reply:
Nope. Doesn't even get applied to type constraints ☹️

Reply:
I fixed this instance. I'll do an audit of the others.